connexion/security.py

```diff
@@ -403,6 +523,126 @@ async def wrapper(request, token, required_scopes):
 
         return wrapper
 
+
+class OIDCSecurityHandlerFactory(AbstractSecurityHandlerFactory):
+    """
+    SecurityHandlerFactory for the OIDC security scheme.
+    """
+
+    # TODO
+
+
+SECURITY_HANDLERS = {
+    # Swagger 2: `type: basic`
+    # OpenAPI 3: `type: http` and `scheme: basic`
+    "basic": BasicSecurityHandlerFactory,
+    # Swagger 2 and OpenAPI 3
+    "apiKey": ApiKeySecurityHandlerFactory,
+    "oauth2": OAuthSecurityHandlerFactory,
+    # OpenAPI 3 only
+    "openIdConnect": OIDCSecurityHandlerFactory,
+    # OpenAPI 3: http schemes
+    "bearer": BearerSecurityHandlerFactory,
+}
+
+
+# TODO: Split up
+#   along security schemes
+#   along 'functionality' -> get_*_func (resolver), security_{deny,passthrough}, and verify_*
+class SecurityHandlerFactory(StandardHandlerFactory):
+    """
+    get_*_func -> _get_function -> get_function_from_name (name=security function defined in spec)
+        (if url defined instead of a function -> get_token_info_remote)
+
+    verify_* -> returns a security wrapper around the security function
+        check_* -> returns a function tasked with doing auth for use inside the verify wrapper
+            check helpers (used outside wrappers): _need_to_add_context_or_scopes
+            the security function
+
+        verify helpers (used inside wrappers): get_auth_header_value, get_cookie_value
+    """
+
+    no_value = AbstractSecurityHandlerFactory.no_value  # TODO
+    required_scopes_kw = "required_scopes"
+    context_kw = "context_"
+    client = None
+
+    def __init__(self, security_handlers: t.Optional[dict] = None) -> None:
+        super().__init__()
+        self.security_handlers = SECURITY_HANDLERS.copy()
+        if security_handlers is not None:
```

Comment:
We still need to expose this to the user. Currently the `SecurityHandlerFactory` is instantiated without passing any options at https://github.com/spec-first/connexion/pull/1671/files#diff-a1af7a3afc2ba04857107ee0179f4de7486723ef00531fa1a3685bcf491e2a78R103.